### PR TITLE
tinyproxy: fix var/etc placement

### LINKED ISSUE #39748: Tinyproxy fail to run due to prefix
Using the latest `tinyproxy 1.8.3`, I get this

```
$ tinyproxy -d
tinyproxy: Could not create file /usr/local/Cellar/tinyproxy/1.8.3/var/log/tinyproxy/tinyproxy.log: No such file or directory
tinyproxy: Could not create file /usr/local/Cellar/tinyproxy/1.8.3/var/run/tinyproxy/tinyproxy.pid: No such file or directory
tinyproxy: Could not create PID file.
```

Suspect `configure` option needs to be updated in the formula.

### PR DESCRIPTION
**Pull request by DomT4**
Closes #39748.

### CODE REVIEW

File: Library/Formula/tinyproxy.rb [hunk 1]
```diff
@@ -11,36 +9,56 @@ class Tinyproxy < Formula
...
   # Fix linking error, via MacPorts: https://trac.macports.org/ticket/27762
   patch :p0 do
     url "https://trac.macports.org/export/83413/trunk/dports/net/tinyproxy/files/patch-configure.diff"
-    sha1 "e946269b681f3ffaa1acd120c93050cb63bfe743"
+    sha256 "414b8ae7d0944fb8d90bef708864c4634ce1576c5f89dd79539bce1f630c9c8d"
   end
 
   def install
+    (var/"log/tinyproxy").mkpath
+    (var/"run/tinyproxy").mkpath
```
Comment (mikemcquaid) [inline Library/Formula/tinyproxy.rb]: Do these need to be in `postinstall` to be kept or does specifying the `localstatedir` below mean there's files in them?

Comment (DomT4) [inline Library/Formula/tinyproxy.rb]: Nope. Nothing installed, they just need to exist before the binary is executed. Have pushed them into `post_install`.

File: Library/Formula/tinyproxy.rb [hunk 2]
```diff
@@ -11,36 +9,58 @@ class Tinyproxy < Formula
...
+    begin
+      pid = fork do
+        exec "#{sbin}/tinyproxy"
+      end
+      sleep 2
+    ensure
+      Process.kill("SIGINT", pid)
+      Process.wait(pid)
+    end
   end
```
Comment (xu-cheng) [inline Library/Formula/tinyproxy.rb]: I think this test will never fail. You may need to test the proxy with `curl`?

Comment (DomT4) [inline Library/Formula/tinyproxy.rb]: A test that never fails? That sounds like something Homebrew *would* want :wink:. Have fixed it. I'm always just hesitant to trust the test-bot to get internal network connections right, It's had major issues before.

**Pull request closed without merging**